### ISSUE TITLE
Programmers Guide -> More topics Users want convered

### ISSUE THREAD

**Issue by slackmoehrle** [chukong/cocos-docs#865]
The API reference should point out which are the deprecated modules and functions (like DrawPrimitives or selectors in Scheduler) and emphasize the freshest methodologies.

The wiki/tutorial segment really needs a rework in certain areas. Scheduler is for example one of the most important concepts and what is being described there doesn't reflect it, it just compares it to a deprecated feature and references one function of the API. Threading doesn't inform anyone what the pitfalls of communicating with the Cocos ui thread are or what the threading model is based around. Asset manager makes no mention of CSLoader, neither does the API reference, which makes a trivial and essential task hard to be acquainted with. HOW TO USE CCLOG is advocating the usage of a deprecated functionality as does many other sections.

When I'll be more familiar with the framework I'd like to actually help with these too but that will take time.

Furthermore there are a LOT of wrongly written English words and statements undermining the integrity of the Docs that must have been in existence for years, why these are ignored is a mystery to me.

**Comment by stevetranby** [comment]
I'll take a look at one or two items, but where do I submit PR's? Is this the repo to submit to now? I hope it is as I don't think changes should be in more than one place for v.Next.

**Comment by slackmoehrle** [comment]
Yes, this repo is where everything should go. We are putting all English docs here, including the Wiki soon. There is always a staging version at slackmoehrle.github.io. That is always the latest. What is in production might not be behind until it is deployed out there.

**Comment by stevetranby** [comment]
I don't think you should have a staging repo separate from this repo. Why not create a separate branch instead? Then the PRs can all be submitted here and worked on here. I guess my point is that I don't want to spend even a second on anything that's already fixed (or in-progress) over on  slackmoehrle.github.io because I was only looking at chukong/cocos-docs?

Main reason I haven't started submitting anything other than simple typo fixes. Everything always feels like it's in flux.

**Comment by slackmoehrle** [comment]
Sorry, what is here: slackmoehrle.github.io.

is built from this cocos-docs repo, the build.sh. 

slackmoehrle.github.io. is just a bunch of HTML files

**Comment by stevetranby** [comment]
Cool, thanks for the info. So do you have a topic that needs work? Something small, but one I can tackle over the next two weeks. I could also do a once over for grammar/typos, but I'm definitely no expert editor :D

**Comment by slackmoehrle** [comment]
One glaring issue for me is that chapters in the Programmers Guide don't all have JavaScript equivs to the c++ code. If that is to much to tackle, grammar and typos is very much appreciated!

**Comment by stevetranby** [comment]
Dang man, 3.5GB for the docs :D

**Comment by slackmoehrle** [comment]
Oh, man, yeah, sorry. I actually that about this on Friday. I did not think of that. We have a lot of docs. Perhaps I could exclude part of the static pages from the repo that wouldn't change (old api-refs) and let the user download them as a .zip if they needed the. That would save 3gb

**Comment by slackmoehrle** [comment]
@steveTrandy - I added a build slim option that skips building the api-ref and PDF and EPUB.
./build.sh --slim. I'll think about how to make this whole process better. It started off simple and then has become complex without taking a step back and thinking about how it is being built. I started some clean up to get rid of the original arrays that were used. I'll get to this in these next revisions I am working on now.
